File: sdks/python/src/utxoiq/resources/billing.py
"""Billing resource for utxoIQ API."""
from ..models import Subscription


class BillingResource:
    """Resource for managing billing and subscriptions."""
    
    def __init__(self, client):
        self.client = client
    
    def get_subscription(self) -> Subscription:
        """
        Get current user subscription information.
        
        Returns:
            Subscription object
        """
        response = self.client.get("/billing/subscription")
        return Subscription(**response.json())
    
    def create_checkout_session(self, tier: str, success_url: str, cancel_url: str) -> dict:
        """
        Create Stripe checkout session for subscription.
        
        Args:
            tier: Subscription tier (pro, power)
            success_url: URL to redirect on success
            cancel_url: URL to redirect on cancel
        
        Returns:
            Dictionary with checkout session URL
        """
        payload = {
            "tier": tier,
            "success_url": success_url,
            "cancel_url": cancel_url
        }
        response = self.client.post("/billing/checkout", json=payload)
        return response.json()
    
    def cancel_subscription(self) -> Subscription:
        """
        Cancel current subscription at period end.
        
        Returns:
            Updated Subscription object
        """
        response = self.client.post("/billing/cancel")
        return Subscription(**response.json())

File: sdks/javascript/src/resources/billing.ts
/**
 * Billing resource for utxoIQ API
 */

import type { AxiosInstance } from 'axios';
import type { Subscription, CreateCheckoutSessionParams } from '../types';

export class BillingResource {
  constructor(private axios: AxiosInstance) {}

  async getSubscription(): Promise<Subscription> {
    const response = await this.axios.get<Subscription>('/billing/subscription');
    return response.data;
  }

  async createCheckoutSession(params: CreateCheckoutSessionParams): Promise<{ url: string }> {
    const response = await this.axios.post<{ url: string }>('/billing/checkout', {
      tier: params.tier,
      success_url: params.successUrl,
      cancel_url: params.cancelUrl,
    });
    return response.data;
  }

  async cancelSubscription(): Promise<Subscription> {
    const response = await this.axios.post<Subscription>('/billing/cancel');
    return response.data;
  }
}
